app/code/Magento/InventoryImportExport/Model/Export/Sources.php

```diff
@@ -0,0 +1,290 @@
+            $sourceIdAttribute->setBackendType('int');
+            $this->_attributeCollection->addItem($sourceIdAttribute);
+
+            /** @var \Magento\Eav\Model\Entity\Attribute $statusIdAttribute */
+            $statusIdAttribute = $this->attributeFactory->create();
+            $statusIdAttribute->setId(SourceItemInterface::STATUS);
+            $statusIdAttribute->setDefaultFrontendLabel(SourceItemInterface::STATUS);
+            $statusIdAttribute->setAttributeCode(SourceItemInterface::STATUS);
+            $statusIdAttribute->setBackendType('int');
+            $statusIdAttribute->setFrontendInput('select');
+            $statusIdAttribute->setSourceModel(StockStatus::class);
+            $this->_attributeCollection->addItem($statusIdAttribute);
+
+            /** @var \Magento\Eav\Model\Entity\Attribute $quantityAttribute */
+            $quantityAttribute = $this->attributeFactory->create();
+            $quantityAttribute->setId(SourceItemInterface::QUANTITY);
+            $quantityAttribute->setBackendType('decimal');
+            $quantityAttribute->setDefaultFrontendLabel(SourceItemInterface::QUANTITY);
+            $quantityAttribute->setAttributeCode(SourceItemInterface::QUANTITY);
+            $this->_attributeCollection->addItem($quantityAttribute);
+        }
+
+        return $this->_attributeCollection;
+    }
+
+    /**
+     * Export process
+     *
+     * @return string
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function export()
+    {
+        $writer = $this->getWriter();
+
+        $columns  = $this->_getHeaderColumns();
+        $writer->setHeaderCols($columns);
+
+        /** @var SourceItemCollection $collection */
+        $collection = $this->sourceItemCollectionFactory->create();
+        $collection->addFieldToSelect($columns);
+
+        $this->applyFilters($collection);
+
+        foreach ($collection->getData() as $data) {
+            unset($data[SourceItem::ID_FIELD_NAME]);
+            $writer->writeRow($data);
+        }
+
+        return $writer->getContents();
+    }
+
+    /**
+     * @param Collection $collection
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    private function applyFilters(Collection $collection)
+    {
+        foreach ($this->retrieveFilterDataFromRequest() as $columnName => $value) {
+            $attributeDefinition = $this->getAttributeCollection()->getItemById($columnName);
+            $type = null;
+            if ($attributeDefinition) {
+                $type = $attributeDefinition->getData('backend_type');
+            }
+
+            $filterMethodName = 'applyFilterFor' . ucfirst($type);
+            if (method_exists($this, $filterMethodName)) {
+                $this->$filterMethodName($collection, $columnName, $value);
+            } else {
+                $this->applyDefaultFilter($collection, $columnName, $value);
+            }
+        }
+    }
+
+    /**
```

Comment:
Both of these refactoring applied will help you to decrease the number of external dependencies for this class

As currently we have a failure in Static Tests:
`The class Sources has a coupling between objects value of 15. Consider to reduce the number of dependencies under 13.`